Src/FluentAssertions/Primitives/StringAssertions.cs

```diff
@@ -1114,6 +1114,86 @@ public AndConstraint<TAssertions> BeNullOrWhiteSpace(string because = "", params
             return new AndConstraint<TAssertions>((TAssertions)this);
         }
 
+        /// <summary>
+        /// Asserts that a string is uppercase.
+        /// </summary>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> BeUpper(string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .ForCondition(Subject?.All(char.IsUpper) == true)
```

Comment:
What if the string is empty?

Reply:
That means it's not upper case. And I've actually asked this question [here](https://github.com/fluentassertions/fluentassertions/issues/1356#issuecomment-659963900).